test/anyone-can-renew-test.js

```diff
@@ -0,0 +1,331 @@
+      value: coin.value,
+      address: coin.address
+    }));
+    mtx.output(0).covenant.type = rules.types.UPDATE;
+    mtx.output(0).covenant.pushHash(nameHash);
+    mtx.output(0).covenant.pushU32(heightBeforeOpen + 1);
+    mtx.output(0).covenant.push(Buffer.alloc(1));
+
+    await alice.fund(mtx, {coins: [coin]});
+    await alice.finalize(mtx, {coins: [coin]});
+    await alice.sign(mtx);
+
+    // Have to add this last because wallet fund clears existing input data
+    const witness = new Stack();
+    witness.pushData(script.encode());
+    mtx.inputs[0].witness.fromStack(witness);
+
+    assert.throws(
+      () => mtx.check(),
+      {message: 'EVAL_FALSE'}
+    );
+  });
+
+  it('should fail to spend without correct action type: TRANSFER', async () => {
+    const mtx = new MTX();
+    mtx.addCoin(coin);
+
+    mtx.addOutput(new Output({
+      value: coin.value,
+      address: coin.address
+    }));
+    mtx.output(0).covenant.type = rules.types.TRANSFER;
+    mtx.output(0).covenant.pushHash(nameHash);
+    mtx.output(0).covenant.pushU32(heightBeforeOpen + 1);
+    mtx.output(0).covenant.pushU8(0);
+    mtx.output(0).covenant.push(Buffer.alloc(20));
+
+    await alice.fund(mtx, {coins: [coin]});
+    await alice.finalize(mtx, {coins: [coin]});
+    await alice.sign(mtx);
+
+    // Have to add this last because wallet fund clears existing input data
+    const witness = new Stack();
+    witness.pushData(script.encode());
+    mtx.inputs[0].witness.fromStack(witness);
+
+    assert.throws(
+      () => mtx.check(),
+      {message: 'EVAL_FALSE'}
+    );
+  });
+
+  it('should spend with correct action type: RENEW - Alice', async () => {
+    const mtx = new MTX();
+    mtx.addCoin(coin);
+
+    mtx.addOutput(new Output({
+      value: coin.value,
+      address: coin.address
+    }));
+    mtx.output(0).covenant.type = rules.types.RENEW;
+    mtx.output(0).covenant.pushHash(nameHash);
+    mtx.output(0).covenant.pushU32(heightBeforeOpen + 1);
+    mtx.output(0).covenant.pushHash(node.chain.tip.hash);
+
+    await alice.fund(mtx, {coins: [coin]});
+    await alice.finalize(mtx, {coins: [coin]});
+    await alice.sign(mtx);
+
+    // Have to add this last because wallet fund clears existing input data
+    const witness = new Stack();
+    witness.pushData(script.encode());
+    mtx.inputs[0].witness.fromStack(witness);
+
+    mtx.check();
+  });
+
+  it('should spend with correct action type: RENEW - Bob', async () => {
+    const mtx = new MTX();
+    mtx.addCoin(coin);
+
+    mtx.addOutput(new Output({
+      value: coin.value,
+      address: coin.address
+    }));
+    mtx.output(0).covenant.type = rules.types.RENEW;
+    mtx.output(0).covenant.pushHash(nameHash);
+    mtx.output(0).covenant.pushU32(heightBeforeOpen + 1);
+    mtx.output(0).covenant.pushHash(node.chain.tip.hash);
+
+    await bob.fund(mtx, {coins: [coin]});
+    await bob.finalize(mtx, {coins: [coin]});
+    await bob.sign(mtx);
+
```

Comment:
Is there a possibility that someone might grief the network by spamming renewals on every name sent here, using dust fee?

Reply:
That is a possibility but renewals in general are rate-limited already by two factors:

1) Renewals must be at least 1 tree interval apart (36 blocks): 

https://github.com/handshake-org/hsd/blob/f2702f873928814d6b5b4bb32985e3163b85f55a/lib/blockchain/chain.js#L1262-L1267

2) Renewals are limited to 600 per block:

https://github.com/handshake-org/hsd/blob/f2702f873928814d6b5b4bb32985e3163b85f55a/lib/protocol/consensus.js#L208

Of course this pull request doesn't change anything - the cat's out of the bag. @tynes and I actually spoke with the project founders about this potential rule-bypass during testnet and there was some discussion about altering `OP_TYPE` to more strictly enforce the _intent_ of the renewal requirement, but ultimately it didn't get implemented.

I suppose there is a potential future risk to users that deploy contracts like this, that in the future the community may deploy a fork and break the pattern but right now, and especially in the use case described by @ca98am79, it seems to be a benefit.